tests/unittests/dataset_tests.py

```diff
@@ -1022,6 +1023,89 @@ def test_iter_samples(self):
             )
         )
 
+    @drop_datasets
+    def test_update_samples(self):
```

Comment:
I don't know if we should be using the "process" backend in unit tests? Multiple processes is nonstandard. Threads too but that has already been broken long ago.